deps/rabbitmq_management/src/rabbit_mgmt_util.erl

```diff
@@ -210,6 +210,14 @@ vhost_from_headers(ReqData) ->
 vhost(ReqData) ->
     rabbit_web_dispatch_access_control:vhost(ReqData).
 
+vhost_or_bad_request(ReqData, Context) ->
```

Comment:
With `resource_exists/2`, this function would be simpler (as it would only return a boolean) and can be made composable with other functions, e.g. those that verify that a queue exists, if this PR decides to go that far.

Reply:
The thing with `resource_exists/2` and `allow_missing_post/2` is that we don't reference that the vhost is the resource missing.

```
< HTTP/1.1 404 Not Found
< content-length: 49
< content-security-policy: script-src 'self' 'unsafe-eval' 'unsafe-inline'; object-src 'self'
< content-type: application/json
< date: Tue, 09 Apr 2024 11:42:21 GMT
< server: Cowboy
< vary: accept, accept-encoding, origin
<
* Connection #0 to host localhost left intact
{"error":"Object Not Found","reason":"Not Found"}
```
Considering what I stated on my [other comment](https://github.com/rabbitmq/rabbitmq-server/pull/10908#discussion_r1557549571) it felt more appropriate to trigger a `not_found` in a way I could make explicit that the vhost was the missing resource.

Maybe a better solution would be to put the call to `rabbit_mgmt_util:not_found` either on the `resource_exists/2` callback, if the vhost does not exist, or on the `allow_missing_post/2` callback.